libraries/botbuilder-repo-utils/src/run.ts

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+export type Success = {
+    _type: 'success';
+    statusCode: number;
+};
+
+export function isSuccess(result: Result): result is Success {
+    return result._type === 'success';
+}
+
+export function success(statusCode = 0): Success {
+    return { _type: 'success', statusCode };
+}
+
+export type Failure = {
+    _type: 'failure';
+    statusCode: number;
+    message?: string;
+};
+
+export function failure(statusCode = 1, message?: string): Failure {
+    return { _type: 'failure', statusCode, message };
+}
+
+export function isFailure(result: Result): result is Failure {
+    return result._type === 'failure';
+}
+
+export type Result = Success | Failure;
+
+export async function run(logic: () => Promise<Result>): Promise<void> {
+    return logic()
+        .then((result) => {
+            if (isFailure(result)) {
+                if (result.message) {
+                    console.error('[error]:', result.message);
+                }
+            }
+
+            process.exit(result.statusCode);
+        })
+        .catch((err) => {
+            console.error(err);
+            process.exit(-1);
```

Comment:
Could you remind me of the significance of `-1` as the process' status code? I believe I've seen this before but I've forgotten what it means in Node packages/tools.

Reply:
```
jpg@fkarakus-ws ~> sh -c 'exit -1;'
jpg@fkarakus-ws ~> echo $?
255
```

It's generally just used to signify "something happened that was totally unexpected". It's not a Node.js specific thing, it's common across many languages and tools.

Reply:
Understood, thanks!